mne/source_estimate.py

```diff
@@ -3346,16 +3350,25 @@ def stc_near_sensors(evoked, trans, subject, distance=0.01, mode='sum',
     subject = _check_subject(None, subject, raise_error=False)
     subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
     if src is None:  # fake a full surface one
-        rrs = [read_surface(op.join(subjects_dir, subject,
-                                    'surf', f'{hemi}.pial'))[0]
-               for hemi in ('lh', 'rh')]
+        _validate_type(surface, str, 'surface', extra='when src is None')
+        rrs = [read_surface(op.join(
+            subjects_dir, subject, 'surf', f'{hemi}.{surface}'))[0]
+            for hemi in ('lh', 'rh')]
```

Comment:
This code is less DRY than a variant of what I proposed because it repeats the concatenation step. Code written this way makes maintenance harder. The flow here is essentially the same, but more DRY:
```
if surface is not None:
    surf_rr = np.concatenate([
        read_surface(op.join(subjects_dir, subject, 'surf',
                     f'{hemi}.{surface}'))[0])
        for hemi in ('lh', 'rh')]
if src is None:
    _validate_type(surface, str, 'surface', extra='when src is None')
    rrs = surf_rr
    src = SourceSpaces([
        dict(rr=rr / 1000., vertno=np.arange(len(rr)), type='surf',
             coord_frame=FIFF.FIFFV_COORD_MRI)
        for rr in rrs])
    keep_all = False
else:
    if surface is not None:
        rrs = surf_rr
    else:
        rrs = np.concatenate([s['rr'][s['vertno']] for s in src])
        if src[0]['coord_frame'] == FIFF.FIFFV_COORD_HEAD:
            rrs = apply_trans(trans, rrs)
    keep_all = True
```

Reply:
Sounds good to me, I was just confused by the first way you wrote it but this looks much more clear